store/root.js

```diff
@@ -0,0 +1,88 @@
+import { observable, action, computed } from 'mobx'
+
+import activities from './activities'
+import User from './user'
+import DepositDates from './deposit-dates'
+import Works from './works'
+
+import apiRequest from 'api'
+
+class Root {
+  @observable user = new User()
+
+  @observable dataProvider = null
+
+  @observable activity = null
+
+  @observable statistics = {
+    metadataCount: 0,
+    fullTextCount: 0,
+    doiCount: 0,
+  }
+
+  @observable works = null
+
+  @observable depositDates = null
+
+  @computed
+  get dataProviders() {
+    return this.user.dataProviders
+  }
+
+  @action async init(dataProviderId, activityPath) {
```

Comment:
I was thinking whether this should be marked as an action or not. Do you have any opinion about it?

[From docs](https://mobx.js.org/refguide/action.html):

> Actions should only, and always, be used on functions that modify state. Functions that just perform look-ups, filters etc should not be marked as actions; to allow MobX to track their invocations.

I haven't found a big advantage in using actions yet. I guess it's good for debugging store state with [Mobx Devtools](https://github.com/mobxjs/mobx-devtools)

Reply:
I marked with `@action` each function that modifies the state according to the doc. If function calls the function that modifies the state, it's an action too. I am not sure about the second but agree to the first one.